### ISSUE TITLE
openmtp is great, but miss image thumbnails (mandatory!)

### ISSUE THREAD

**Issue by chjan** [ganeshrvel/openmtp#242]
Can you please consider adding thumbnails as default in the view of the mobile device? It is VERY hard to search&find the correct picture on the device, for transferral to a mac laptop, when all you get is a dated number of the photo.

Many thanks for your consideration!

**Comment by resuna** [comment]
Suggested workaround: transfer all the files and delete the ones you don't want, since it has to transfer the files to generate the thumbnails anyway. At least it shouldn't be the default because it would significantly slow down directory listings.

**Comment by ganeshrvel** [comment]
@resuna is correct. Generating thumbnail isnt an easy task and will slow down the directory listing significantly.

**Comment by chjan** [comment]
I accept that slowness because I would be granted strongly enhanced usability!

**Comment by KingSupernova31** [comment]
Yeah this tool is just not particularly useful without thumbnails. (Its precursors Android File Transfer and Handshaker both supported them.)

**Comment by ganeshrvel** [comment]
I understand the frustration, but fetching and maintaining the thumbnails out of the mtp device is a different challenge altogether as it could slow down the UX. Thumbs require careful handling, and involves proper cache invalidation.

I’m currently rebuilding the app from scratch to make it more lightweight, smooth, and feature-rich. Protocols for retrieving thumbnails have been added. Will try to get it working on the UI as well.
